ast/src/main/java/com/graphicsfuzz/common/ast/type/BasicType.java

```diff
@@ -447,6 +447,40 @@ public boolean isMatrix() {
     return allMatrixTypes().contains(this);
   }
 
+  /**
```

Comment:
When #648 and this are merged I can refactor this function to use the helper makeMatrixType() function added in that PR.